chips/msp432/src/cs.rs

```diff
@@ -0,0 +1,275 @@
+        HSMCLK_READY OFFSET(26) NUMBITS(1),
+        SMCLK_READY OFFSET(27) NUMBITS(1),
+        BCLK_READY OFFSET(28) NUMBITS(1)
+    ],
+    // interrupt enable register
+    CSIE [
+        // LFXT oscillator fault flag
+        LFXTIE OFFSET(0) NUMBITS(1),
+        // HFXT oscillator fault flag
+        HFXTIE OFFSET(1) NUMBITS(1),
+        // DCO external resistor open circuit fault flag
+        DCOR_OPNIE OFFSET(6) NUMBITS(1),
+        // LFXT start fault counter
+        FCNTLFIE OFFSET(8) NUMBITS(1),
+        // HFXT start fault counter
+        FCNTHFIE OFFSET(9) NUMBITS(1)
+    ],
+    // interrupt flag register
+    CSIFG [
+        // LFXT oscillator fault flag
+        LFXTIFG OFFSET(0) NUMBITS(1),
+        // HFXT oscillator fault flag
+        HFXTIFG OFFSET(1) NUMBITS(1),
+        // DCO external resistor open circuit fault flag
+        DCOR_OPNIFG OFFSET(6) NUMBITS(1),
+        // LFXT start fault counter
+        FCNTLFIFG OFFSET(8) NUMBITS(1),
+        // HFXT start fault counter
+        FCNTHFIFG OFFSET(9) NUMBITS(1)
+    ],
+    // interrupt clear register
+    CSCLRIFG [
+        // LFXT oscillator fault flag
+        LFXTIFG OFFSET(0) NUMBITS(1),
+        // HFXT oscillator fault flag
+        HFXTIFG OFFSET(1) NUMBITS(1),
+        // DCO external resistor open circuit fault flag
+        DCOR_OPNIFG OFFSET(6) NUMBITS(1),
+        // LFXT start fault counter
+        FCNTLFIFG OFFSET(8) NUMBITS(1),
+        // HFXT start fault counter
+        FCNTHFIFG OFFSET(9) NUMBITS(1)
+    ],
+    // interrupt set/assert register
+    CSSETIFG [
+        // LFXT oscillator fault flag
+        SET_LFXTIFG OFFSET(0) NUMBITS(1),
+        // HFXT oscillator fault flag
+        SET_HFXTIFG OFFSET(1) NUMBITS(1),
+        // DCO external resistor open circuit fault flag
+        SET_DCOR_OPNIFG OFFSET(6) NUMBITS(1),
+        // LFXT start fault counter
+        SET_FCNTLFIFG OFFSET(8) NUMBITS(1),
+        // HFXT start fault counter
+        SET_FCNTHFIFG OFFSET(9) NUMBITS(1)
+    ],
+    // DCO external resistor calibration 0 register
+    CSDCOERCAL0 [
+        // DCO temperature compensation calibration
+        DCO_TCCAL OFFSET(0) NUMBITS(1),
+        // DCO frequency calibration for DCO frequency range (DCORSEL) 0 to 4
+        DCO_FCAL_RSEL04 OFFSET(16) NUMBITS(10)
+    ],
+     // DCO external resistor calibration 1 register
+    CSDCOERCAL1 [
+        // DCO frequency calibration for DCO frequency range (DCORSEL) 5
+        DCO_FCAL_RSEL5 OFFSET(0) NUMBITS(10)
+    ]
+];
+
+pub struct ClockSystem {
+    registers: StaticRef<CsRegisters>,
+}
+
+impl ClockSystem {
+    const fn new() -> ClockSystem {
+        ClockSystem { registers: CS_BASE }
+    }
+
+    fn unlock_registers(&self) {
+        self.registers.key.modify(CSKEY::KEY.val(KEY));
+    }
+
+    fn lock_registers(&self) {
+        // Every value except KEY written to the key register will perform the lock
+        self.registers.key.modify(CSKEY::KEY.val(0));
+    }
```

Comment:
This is okay, but it would be better to use the [`PeripheralManager`](https://docs.tockos.org/kernel/common/peripherals/index.html) `before/after_peripheral_access` hooks, which allow an RAII-style interface that will automatically lock and unlock correctly.

Reply:
Ok, I will try to convert it to this model!

Reply:
I have now implemented the `PeripheralManagement` trait for the clock system, I hope I did it correct!